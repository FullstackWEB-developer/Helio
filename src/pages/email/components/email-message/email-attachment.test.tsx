import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render,fireEvent,waitForElement} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import EmailAttachment from './email-attachment';
import Router from "react-router-dom";
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("Email Attachment tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders email-attachment correctly", async () => {
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <EmailAttachment attachment={{fileName: "s", mimeType: "s"}} messageId={"s"} />
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders email-attachment click", async () => {
        const {getByTestId} = render(<TestWrapper mockState={mockState}>
            <EmailAttachment attachment={{fileName: "s", mimeType: "s"}} messageId={"s"} />
        </TestWrapper>);
        
        fireEvent.click(getByTestId("download-button"))
    });

    it("renders email-attachment mouse-over", async () => {
        const {getByTestId} = render(<TestWrapper mockState={mockState}>
            <EmailAttachment attachment={{fileName: "s", mimeType: "s"}} messageId={"s"} />
        </TestWrapper>);
        
        fireEvent.mouseOver(getByTestId("download-button"))
    });

    it("renders email-attachment mouse-out", async () => {
        const {getByTestId} = render(<TestWrapper mockState={mockState}>
            <EmailAttachment attachment={{fileName: "s", mimeType: "s"}} messageId={"s"} />
        </TestWrapper>);
        
        fireEvent.mouseOut(getByTestId("download-button"))
    });
})