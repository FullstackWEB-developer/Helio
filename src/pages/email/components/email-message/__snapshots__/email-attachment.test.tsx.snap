// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Attachment tests renders email-attachment correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center email-attachment-thumbnail w-60 p-4 mb-4 cursor-pointer body3-medium attachment-file-name h-16"
    data-testid="download-button"
  >
    <div
      class="pr-2"
    >
      <svg
        class="icon-large-40 "
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
          id="Layer_2"
        >
          <g
            id="Attachment_generic_48px"
          >
            <rect
              class="attachment-1 fill-default"
              height="48"
              width="48"
            />
            <polygon
              class="attachment-2 fill-default"
              points="26.99 5 10.17 5 10.17 26.96 10.17 43.48 38.27 43.48 38.27 26.96 38.27 16.51 26.99 5"
            />
            <path
              class="attachment-2 fill-default"
              d="M27,5V16.28H38.26"
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      class="truncate"
    >
      s
    </div>
  </div>
</DocumentFragment>
`;
